### PR TITLE
A0-2250: Fix various inadequate code practices

### PR DESCRIPTION
**Pull request by jalooc**

### CODE REVIEW

File: packages/extension-ui/src/Popup/CreateAccount/SaveMnemonic.tsx [hunk 1]
```diff
@@ -47,10 +47,12 @@ function SaveMnemonic({ className, onNextStep, onPreviousStep, seed }: Props): R
...
           ))}
         </div>
-        <CopyToClipboard text={seed}>
+        <CopyToClipboard
+          onCopy={_onCopy}
+          text={seed}
+        >
           <Button
             className='copy-button'
-            onClick={_onCopy}
```
Comment (jalooc) [inline packages/extension-ui/src/Popup/CreateAccount/SaveMnemonic.tsx:53]: The `_onCopy` function does nothing apart of showing a toast - that was really weird, considering that copying... works regardless.

Turns out `CopyToClipboard` adds the copyping handler to the first child, and also supplies a _success callback_ prop, so using `onClick` on that child to announce success was just incorrect.

File: packages/extension-ui/src/Popup/Export.tsx [hunk 2]
```diff
@@ -90,14 +88,6 @@ function Export({
             type='password'
             value={pass}
           />
-          {pass.length < MIN_LENGTH && pass !== '' && (
```
Comment (jalooc) [inline packages/extension-ui/src/Popup/Export.tsx:93]: Doesn't really make sense to show password validation errors during actual usage of a password (as opposed to during creation).

File: packages/extension-ui/src/Popup/CreateAccount/SaveMnemonic.tsx [hunk 3]
```diff
@@ -47,10 +47,12 @@ function SaveMnemonic({ className, onNextStep, onPreviousStep, seed }: Props): R
             />
           ))}
         </div>
-        <CopyToClipboard text={seed}>
+        <CopyToClipboard
+          onCopy={_onCopy}
```
Comment (youPickItUp) [inline packages/extension-ui/src/Popup/CreateAccount/SaveMnemonic.tsx:51]: callback signature is `onCopy(text, result)` where result is a boolean indicating is copying was successful. Not sure when can it fails, but perhaps it's worth handling?

**Review by youPickItUp — APPROVED** [review]

File: packages/extension-ui/src/Popup/Signing/TransactionIndex.tsx [hunk 4]
```diff
@@ -23,8 +23,9 @@ function TransactionIndex({
   index,
   onNextClick,
   onPreviousClick,
-  totalItems
+  totalItems: unsanitizedTotalItems
 }: Props): React.ReactElement<Props> {
+  const totalItems = unsanitizedTotalItems < 0 ? 0 : unsanitizedTotalItems;
```
[the remark this answers is not in the document]
Comment (jalooc) [reply packages/extension-ui/src/Popup/Signing/TransactionIndex.tsx:28]: I think they didn't mean integer overflows, but rather _semantical_ overflow of e.g. `index` becoming a value lower than `0` for which the component doesn't make sense.

I think I'll keep it this time for the report purpose, but yes - generally I agree with you that this kind of validation seems tad exaggerated.

File: packages/extension-ui/src/Popup/CreateAccount/SaveMnemonic.tsx (hunk 3 above)
Comment (jalooc) [reply packages/extension-ui/src/Popup/CreateAccount/SaveMnemonic.tsx:51]: Missed that, thanks! Will fix.

**Pull request merged**